splainer-search.js

```diff
@@ -1037,6 +1038,62 @@ angular.module('o19s.splainer-search')
 
 'use strict';
 
+angular.module('o19s.splainer-search')
+  .service('queryTemplateSvc', function queryTemplateSvc() {
+    var self      = this;
+    self.hydrate = hydrate;
+
+    var defaultConfig = {
+      encodeURI: false,
+      defaultKw: '""',
+    };
+
+    function encode(queryPart, config) {
+      if (config.encodeURI) {
+        return encodeURIComponent(queryPart);
+      } else {
+        return queryPart;
+      }
+    }
+
+
+    function keywordMapping(queryText) {
+      var queryTerms    = queryText.split(/[ ,]+/);
+      var maxKeywords = 10;
+      var numTerms = queryTerms.length;
+      for (var i = numTerms; i < maxKeywords; i++) {
+        queryTerms.push(null);
+      }
+      return queryTerms;
+    }
+
+    function hydrate(template, queryText, config) {
+      if (!config) {
+        config = defaultConfig;
+      }
+
+      if (queryText === null || angular.isUndefined(queryText)) {
+        return template;
+      }
+
+      var replaced  = template.replace(/#\$query##/g, encode(queryText, config));
+      var idx = 0;
+      angular.forEach(keywordMapping(queryText), function(queryTerm) {
```

Comment:
I see what you mean

On Wed, Dec 2, 2015 at 10:10 AM, Youssef Chaker <notifications@github.com>
wrote:

> In splainer-search.js
> <https://github.com/o19s/splainer-search/pull/22#discussion_r46425082>:
>
> > +      }
> > +      return queryTerms;
> > +    }
> > +
> > +    function hydrate(template, queryText, config) {
> > +      if (!config) {
> > +        config = defaultConfig;
> > +      }
> > +
> > +      if (queryText === null || angular.isUndefined(queryText)) {
> > +        return template;
> > +      }
> > +
> > +      var replaced  = template.replace(/#\$query##/g, encode(queryText, config));
> > +      var idx = 0;
> > +      angular.forEach(keywordMapping(queryText), function(queryTerm) {
>
> you would still split the original query string into an array, and then do
> terms[i], if that's undefined you replace it with defaultKw like you
> already do.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/o19s/splainer-search/pull/22/files#r46425082>.
>